controllers/ibmpowervsmachinetemplate_controller.go

```diff
@@ -0,0 +1,87 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/go-logr/logr"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/cluster-api/util/patch"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	infrav1beta1 "sigs.k8s.io/cluster-api-provider-ibmcloud/api/v1beta1"
+)
+
+// IBMPowerVSMachineTemplateReconciler reconciles a IBMPowerVSMachineTemplate object.
+type IBMPowerVSMachineTemplateReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+func (r *IBMPowerVSMachineTemplateReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&infrav1beta1.IBMPowerVSMachineTemplate{}).
+		Complete(r)
+}
+
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=ibmpowervsmachinetemplates,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=ibmpowervsmachinetemplates/status,verbs=get;update;patch
+
+func (r *IBMPowerVSMachineTemplateReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := r.Log.WithValues("ibmpowervsmachinetemplate", req.NamespacedName)
+
+	var machineTemplate infrav1beta1.IBMPowerVSMachineTemplate
+	if err := r.Get(ctx, req.NamespacedName, &machineTemplate); err != nil {
+		logger.Error(err, "unable to fetch ibmpowervsmachinetemplate")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	helper, err := patch.NewHelper(&machineTemplate, r.Client)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to init patch helper: %w", err)
+	}
+
+	capacity := getIBMPowerVSMachineCapacity(machineTemplate)
+	logger.V(3).Info("calculated capacity for machine template", "capacity", capacity)
+	if reflect.DeepEqual(machineTemplate.Status.Capacity, capacity) != true {
+		machineTemplate.Status.Capacity = capacity
+		if err := helper.Patch(ctx, &machineTemplate); err != nil {
+			if !apierrors.IsNotFound(err) {
+				logger.Error(err, "failed to patch machineTemplate")
+				return ctrl.Result{}, err
+			}
+		}
+	}
+	logger.V(3).Info("machine template status", "status", machineTemplate.Status.Capacity)
+	return ctrl.Result{}, nil
+}
+
+func getIBMPowerVSMachineCapacity(machineTemplate infrav1beta1.IBMPowerVSMachineTemplate) corev1.ResourceList {
+	capacity := make(corev1.ResourceList)
+	capacity[corev1.ResourceMemory] = resource.MustParse(machineTemplate.Spec.Template.Spec.Memory)
+	capacity[corev1.ResourceCPU] = resource.MustParse(machineTemplate.Spec.Template.Spec.Processors)
```

Comment:
Usually in powervs Processors value is mapped to multiples of SMT to the actual vcpus in the linux vm, hence need that logic here while calculating the capacity!

Reply:
updated the  logical cpus